controller/test/frame_detector/frame_detector_test.cpp

```diff
@@ -0,0 +1,144 @@
+#include "frame_detector.h"
+#include "proto_traits.h"
+#include "soft_rx_buffer.h"
+#include "gtest/gtest.h"
+
+constexpr uint32_t LEN = 20;
+
+constexpr FrameDetector<SoftRxBuffer<LEN>, LEN>::State STATE_LOST =
+    FrameDetector<SoftRxBuffer<LEN>, LEN>::State::LOST;
+constexpr FrameDetector<SoftRxBuffer<LEN>, LEN>::State STATE_WAIT_START =
+    FrameDetector<SoftRxBuffer<LEN>, LEN>::State::WAIT_START;
+constexpr FrameDetector<SoftRxBuffer<LEN>, LEN>::State STATE_RX_FRAME =
+    FrameDetector<SoftRxBuffer<LEN>, LEN>::State::RX_FRAME;
+
+constexpr uint8_t MARK = static_cast<uint8_t>('.');
+
+SoftRxBuffer<LEN> rx_buf('.');
+
+TEST(FrameDetector, TrivialRx) {
+  FrameDetector<SoftRxBuffer<LEN>, LEN> frame_detector(rx_buf);
+  EXPECT_TRUE(frame_detector.Begin());
+  ASSERT_EQ(STATE_LOST, frame_detector.get_state());
+  EXPECT_FALSE(frame_detector.is_frame_available());
+
+  rx_buf.PutByte(' ');
+  ASSERT_EQ(STATE_LOST, frame_detector.get_state());
+  EXPECT_FALSE(frame_detector.is_frame_available());
+
+  rx_buf.PutByte(MARK);
+  ASSERT_EQ(STATE_WAIT_START, frame_detector.get_state());
```

Comment:
E.g. why is this WAIT_START - is it because the marker after a LOST is interpreted as an end of frame marker rather than start of frame, and the preceding frame is discarded?

Reply:
Yes, marker in the LOST condition is interpreted as an END marker.